File: packages/contracts/src/nfts.rs
use scrypto::prelude::*;
use crate::enums::LoanStatus;

#[derive(ScryptoSbor, NonFungibleData, Clone)]
pub struct LoanBorrowerNFT {
    pub component: ComponentAddress,
    pub receiver: ComponentAddress,
    pub amount: Decimal,
    pub total_amount: Decimal,
    pub duration: i64,
    pub apr: Decimal,
    #[mutable]
    pub maturity_date: Option<UtcDateTime>,
    #[mutable]
    pub closed_date: Option<UtcDateTime>,
    #[mutable]
    pub status: LoanStatus,
}

#[derive(ScryptoSbor, NonFungibleData, Clone)]
pub struct LoanLenderNFT {
    pub component: ComponentAddress,

    #[mutable]
    pub status: LoanStatus,

    // pub loan_borrower_nft_id: NonFungibleLocalId,
    // #[mutable]
    // pub status: Status,
}

File: packages/web/src/types/nfts.ts
export interface NonFungibleToken<T> {
  id: string
  type: string
  data: T
}

export interface BorrowerNFTData {
  component: string
  receiver: string
  amount: number
  total_amount: number
  duration: number
  apr: number
  maturity_date: string
  closed_date: string
  status: string
}

export interface LenderNFTData {
  component: string
  status: string
}

export type BorrowerNFT = NonFungibleToken<BorrowerNFTData>

export type LenderNFT = NonFungibleToken<BorrowerNFTData>
